cmd/nvidia-dra-controller/imex.go

```diff
@@ -312,3 +310,33 @@ func (m *ImexManager) cleanupResourceSlices() error {
 
 	return nil
 }
+
+// getOffset set the offset where an imex domain's channels should start counting from.
+func (i *imexDomainOffsets) getOffset(addedDomainsCh string) (int, error) {
+	var offset int
+	id := strings.Split(addedDomainsCh, ".")
+
+	// Check if the IMEX domain is already in the map.
+	if _, ok := (*i)[id[0]]; !ok {
+		(*i)[id[0]] = make(map[string]int)
+	}
+
+	// Check if the clique is already in the map.
+	if _, ok := (*i)[id[0]][id[1]]; !ok {
+		offset = (len((*i)[id[0]]) * ResourceSliceImexChannelLimit)
+		if offset >= DriverImexChannelLimit {
+			return 0, fmt.Errorf("error adding IMEX domain %s: channel limit reached", id[0])
+		}
+		(*i)[id[0]][id[1]] = offset
+	}
+
+	return offset, nil
```

Comment:
This should take the ResourceSliceImexChannelLimit and DriverImexChannelLimit as parameters not assume it knows these values.